--- /Users/justinsmalley/Codebase/influx/influxdb/tsdb/internal/meta.proto	2023-10-24 10:00:00.000000000 -0400
+++ /Users/justinsmalley/Codebase/influx/influxdb/tsdb/internal/meta.proto	2023-10-24 10:00:00.000000000 -0400
@@ -62,3 +62,19 @@
 message FieldMappingSet {
     repeated FieldMappingMeasurement Measurements = 1;
 }
+
+enum MeasurementMappingState {
+    MEASUREMENT_MAPPING_STATE_ACTIVE = 0;
+    MEASUREMENT_MAPPING_STATE_DELETED = 1;
+    MEASUREMENT_MAPPING_STATE_RENAMED = 2;
+}
+
+message MeasurementMapping {
+    required string UserName = 1;
+    required string InternalName = 2;
+    required int64 Version = 3;
+    required MeasurementMappingState State = 4;
+}
+
+message MeasurementMappingSet {
+    repeated MeasurementMapping Mappings = 1;
+}
